test/function/samples/call-marked-pure/_config.js

```diff
@@ -0,0 +1,20 @@
+const assert = require('assert');
+
+module.exports = {
+	description: 'external function calls marked with pure comment do not have effects',
+	options: {
+		external: ['socks']
+	},
+	context: {
+		require(id) {
+			if (id === 'socks') {
+				return () => {
+					throw new Error('Not all socks were removed.');
```

Comment:
The emoji is lovely but since this is supposed to be a functional test, I would want it to throw an error when things happen that are not supposed to happen. With this assertion, the source-code check might probably be moot.